examples/large_models/inferentia2/util/inf2_save_split_checkpoints.py

```diff
@@ -17,7 +20,7 @@ def create_directory_if_not_exists(path_str: str) -> str:
         raise NotADirectoryError(path_str)
 
 
-def amp_callback(model: OPTForCausalLM, dtype: torch.dtype) -> None:
+def opt_amp_callback(model: OPTForCausalLM, dtype: torch.dtype) -> None:
```

Comment:
Why AMP can only be applied on OPT model?

Reply:
Updated model loading to include AMP:

```
        self.model = LlamaForSampling.from_pretrained(
            model_dir,
            batch_size=self.mb_handle.micro_batch_size,
            amp=amp,
            tp_degree=tp_degree,
        )
```